src/iterative_sorting/iterative_sorting.py

```diff
@@ -1,30 +1,42 @@
-# TO-DO: Complete the selection_sort() function below 
-def selection_sort( arr ):
+# TO-DO: Complete the selection_sort() function below
+def selection_sort(arr):
     # loop through n-1 elements
-    for i in range(0, len(arr) - 1):
-        cur_index = i
-        smallest_index = cur_index
+    for i in range(0, len(arr)):
+        smallest_index = i
+        for j in range(i + 1, len(arr) - 1):
+            if arr[j] < arr[smallest_index]:
+                smallest_index = j
+        temp = arr[smallest_index]
+        arr[smallest_index] = arr[i]
+        arr[i] = temp
+        # for j in range(i + 1, len(arr) - 1):
+        #     if arr[j] < arr[smallest_index]:
+        #         smallest_index = j
+
+        # temp = arr[smallest_index]
+        # arr[smallest_index] = arr[cur_index]
+        # arr[cur_index] = temp
         # TO-DO: find next smallest element
-        # (hint, can do in 3 loc) 
-             
-
-
+        # (hint, can do in 3 loc)
 
         # TO-DO: swap
 
-
-
-
     return arr
 
 
 # TO-DO:  implement the Bubble Sort function below
-def bubble_sort( arr ):
+def bubble_sort(arr):
+    for i in range(0, len(arr) - 1):
+        for j in range(0, len(arr) - 1):
+            if arr[j] > arr[j + 1]:
+                temp = arr[j]
+                arr[j] = arr[j + 1]
+                arr[j + 1] = temp
 
```

Comment:
interesting implementation of bubble sort, you may want to look into a bubble sort that has a flag for a slightly better version, your solution is correct, but i think your run time will always be worst case, because it checks everything all the time, an optimized bubble sort is also sorting on the way back as well.